docs/2.5. APIs - Query Parameters.md

```diff
@@ -317,160 +346,180 @@ In this case, assume that the most recently created or updated Node held in the
 
 ### Invalid Examples
 
 **Invalid Example 1: Duplicate Query Parameters**
 
 ***Request***
-```
+
+```http
 GET /x-nmos/query/v1.0/flows?tags.location=Salford&tags.location=London
 ```
 
 ***Response***
-* This query specifies two matching query parameters with different values. As different programming frameworks may not make both of these values available, this is an invalid query for which the response is not defined and may vary by implementation. If this sort of query is required, it is suggested that Resource Query Language is used (see below).
 
-## Advanced (RQL) Queries (Optional)
+- This query specifies two matching query parameters with different values. As different programming frameworks might not make both of these values available, this is an invalid query for which the response is not defined and might vary by implementation. The next section provides a means of achieving the desired query.
+
+## Advanced (RQL) Queries (OPTIONAL)
 
 Query APIs MAY support Resource Query Language (RQL, <https://github.com/persvr/rql/>) queries against their API resources where the 'rql' trait is specified in the RAML documentation. A 501 HTTP status code MUST be returned where a RAML query is attempted using RQL functions or operators which are not supported by a Query API.
 
-RQL should be formatted in the normalised form as opposed to using FIQL syntax, and passed via the query string using '?query.rql=...'.
+RQL SHOULD be formatted in the normalised form as opposed to using FIQL syntax, and passed via the query string using '?query.rql=...'.
 
-Querying within objects and arrays should be performed as in the Basic Queries case by using the '.' separator.
+Querying within objects and arrays SHOULD be performed as in the Basic Queries case by using the '.' separator.
 
 When an RQL query is specified, the Basic Queries format MAY be ignored in order to simplify implementation.
 
-Note that as RQL permits the definition of very complex queries, the server may return a 400 error code to indicate that it is refusing to action the client's request due to the query's complexity.
+Note that as RQL permits the definition of very complex queries, the server MAY return a 400 error code to indicate that it is refusing to action the client's request due to the query's complexity.
 
 ### Constraints
 
-Not all RQL operators will be suitable for use with a Query API. It is suggested that the following operators are supported, however these are not mandated and implementations may choose to support their own subset.
+Not all RQL operators will be suitable for use with a Query API. It is suggested that the following operators are supported, however these are not mandated and implementations MAY choose to support their own subset.
```

Comment:
FWIW, I'm thinking about proposing a BCP that defines a useful RQL subset much more tightly.

Reply:
That would be a good thing. As things stand there is opportunity for non-interoperability here!

Reply:
See also #20